riscv/src/compiler.rs

```diff
@@ -520,7 +526,11 @@ fn preamble(degree: u64, coprocessors: &CoProcessors, with_bootloader: bool) ->
 
     // Except for the last row, if m_change is 1, then m_addr has to increase,
     // if it is zero, m_step has to increase.
-    (1 - LAST) { m_change * (m_addr' - m_addr) + (1 - m_change) * (m_step' - m_step) } in POSITIVE;
+    // `m_diff_upper * 2**16 + m_diff_lower` has to be equal to the difference **minus one**.
+    // Since we know that both m_addr and m_step can only be 32-Bit, this enforces that
+    // the values are strictly increasing.
+    col diff = (m_change * (m_addr' - m_addr) + (1 - m_change) * (m_step' - m_step));
+    (1 - LAST) * (diff - 1 - m_diff_upper * 2**16 - m_diff_lower) = 0;
```

Comment:
`m_diff_upper * 2**16` isn't compiled to estark as a constraint, and is solved before that, right?

Reply:
I'm not sure what you mean?

Reply:
this constraint is syntactically cubic

Reply:
So the `2**16` is evaluated to a constant, so it's just a multiplication by a constant, which doesn't increase the degree. So `diff - 1 - m_diff_upper * 2**16 - m_diff_lower` is degree 1 and the entire term degree 2, no?

Reply:
yea I agree I just don't know how estark handles `m_diff_upper * 2**16`, whether degree 1 or 2